test/sql2dsql/core_test.clj

```diff
@@ -1,146 +1,412 @@
 (ns sql2dsql.core-test
   (:require
-   [clojure.test :refer :all]
-   [sql2dsql.core :refer [->dsql]]))
+    [clojure.test :refer :all]
+    [sql2dsql.core :refer [->dsql]]))
 
 (defn parse [sql & params]
-  (-> (apply ->dsql (cons sql params))
-      first))
+  (try
+    (-> (apply ->dsql (cons sql params))
+        first)
+    (catch Exception e
+      (println (str "Error parsing SQL: " sql))
+      (println (str "Exception: " (.getMessage e)))
+      nil)))
 
+(defmacro test-sql
+  ([num sql expected]
+   `(test-sql ~num ~sql [] ~expected))
+  ([num sql params expected]
+   `(testing (str "Test " ~num ": " ~sql)
+      (let [result# (apply parse ~sql ~params)]
+        (is (= result# ~expected)
+            (str "\nTest " ~num " failed!"
```

Comment:
It's better to use https://clojuredocs.org/clojure.test/do-report
[Example](https://github.com/HealthSamurai/matcho/blob/e651ea6816af421cf22bc0fd2131a5b44ff0d642/src/matcho/core.clj#L129)